android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageViewModel.kt

```diff
@@ -89,7 +51,10 @@ class CreateListingPageViewModel @Inject constructor(
 				fullAddress = it,
 				addressLine = if (splitAddress.isNotEmpty()) splitAddress[0] else "",
 				addressCity = if (splitAddress.size >= 2) splitAddress[1] else "",
-				addressPostalCode = if (splitAddress.size >= 3) splitAddress[2] else "",
+				addressPostalCode = if (
+					splitAddress.size >= 3 &&
+					splitAddress[2].split(" ").size == POSTAL_CODE_LENGTH
+				) splitAddress[2] else "",
```

Comment:
create listing api doesn't like when postal code isn't formatted like `"A1B 2C3"` and since autocomplete gives `"ON"`, postal code will be `""` when using autocomplete